app/src/main/java/ru/yandex/practicum/contacts/presentation/filter/model/FilterContactTypeUi.java

```diff
@@ -2,11 +2,25 @@
 
 import androidx.annotation.NonNull;
 
-public class FilterContactTypeUi {
+import ru.yandex.practicum.contacts.presentation.base.ListDiffInterface;
+
+public class FilterContactTypeUi implements ListDiffInterface<FilterContactTypeUi> {
 
     private final FilterContactType contactType;
     private final boolean selected;
 
+    private FilterContactTypeUi t;
+
+
+    @Override
+    public boolean theSameAs(@NonNull FilterContactTypeUi t){
+        return this.t.hashCode() == t.hashCode();
```

Comment:
В данном случае будет NPE, так как `this.t` нигде не инициализируется. В целом переменная `this.t` не нужна, у тебя есть аргумент функции `t` и есть ссылка на текущий объект `this` - нужно сравнить их поля `contactType`. Корректной реализацией будет:
```suggestion
        return this.getContactType() == t.getContactType();
```

Ниже я напишу так же комментарии, там аналогичная проблема как тут, но напишу только исправления